app/src/main/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsFragment.kt

```diff
@@ -240,30 +243,43 @@ class TransactionDetailsFragment : BaseViewBindingFragment<FragmentTransactionDe
                 val outgoing = txInfo.direction == TransactionDirection.OUTGOING
                 val address = if (outgoing) txInfo.recipient else txInfo.sender
 
-                val txType = if (txInfo.direction == TransactionDirection.INCOMING) {
+                val txType = if (!outgoing) {
                     TxType.TRANSFER_INCOMING
                 } else {
                     TxType.TRANSFER_OUTGOING
                 }
+                val addressUri = when (val info = if (outgoing) txInfo.recipientInfo else txInfo.senderInfo) {
+                    is AddressInfoData.Remote -> info.addressLogoUri
+                    else -> null
+                }
+                val addressName = when (val info = if (outgoing) txInfo.recipientInfo else txInfo.senderInfo) {
```

Comment:
this check should not be needed. You should be able to access name directly from AddressInfoData

Reply:
So `TransactionInfoViewData.Transfer` should not contain `recipientInfo` and `senderInfo` but only `addressInfo`?

Reply:
Done in 6345075b9bb6e5436e7341c238fb12b3eb0d65f9